tests/equality/equality.cpp

```diff
@@ -0,0 +1,995 @@
+
+    v1->setInitialValue(v3);
+    v2->setInitialValue(v4);
+
+    v1->setInterfaceType(libcellml::Variable::InterfaceType::PUBLIC);
+    v2->setInterfaceType(libcellml::Variable::InterfaceType::PUBLIC);
+
+    v1->setUnits(u1);
+    v2->setUnits(u2);
+
+    EXPECT_TRUE(v1->equal(v2));
+}
+
+TEST(Equality, variableNotEqualByUnits)
+{
+    libcellml::VariablePtr v1 = libcellml::Variable::create("variable");
+    libcellml::VariablePtr v2 = libcellml::Variable::create("variable");
+
+    libcellml::UnitsPtr u1 = libcellml::Units::create("m_per_s");
+
+    v1->setUnits(u1);
+
+    EXPECT_FALSE(v1->equal(v2));
+}
+
+TEST(Equality, variableNotEqualByUnitsOther)
+{
+    libcellml::VariablePtr v1 = libcellml::Variable::create("variable");
+    libcellml::VariablePtr v2 = libcellml::Variable::create("variable");
+
+    libcellml::UnitsPtr u2 = libcellml::Units::create("m_per_s");
+
+    v2->setUnits(u2);
+
+    EXPECT_FALSE(v1->equal(v2));
+}
+
+TEST(Equality, unitsEqualByNameOnly)
+{
+    libcellml::UnitsPtr u1 = libcellml::Units::create("units");
+    libcellml::UnitsPtr u2 = libcellml::Units::create("units");
+
+    EXPECT_NE(u1, u2);
+    EXPECT_TRUE(u1->equal(u2));
+    EXPECT_TRUE(libcellml::Units::equivalent(u1, u2));
+}
+
+TEST(Equality, unitsNotEqualByNameOnly)
+{
+    libcellml::UnitsPtr u1 = libcellml::Units::create("unitsA");
+    libcellml::UnitsPtr u2 = libcellml::Units::create("unitsB");
+
+    EXPECT_NE(u1, u2);
+    EXPECT_FALSE(u1->equal(u2));
+    EXPECT_FALSE(libcellml::Units::equivalent(u1, u2));
+}
+
+TEST(Equality, unitsEqualAllTestedConditionsVariant1)
+{
+    libcellml::UnitsPtr u1 = libcellml::Units::create("unitsA");
+    libcellml::UnitsPtr u2 = libcellml::Units::create("unitsA");
+
+    libcellml::ImportSourcePtr is = libcellml::ImportSource::create();
+
+    u1->setImportReference("some_location");
+    u2->setImportReference("some_location");
+
+    u1->setSourceUnits(is, "ampere");
+    u2->setSourceUnits(is, "ampere");
+
+    u1->addUnit("meter");
+    u2->addUnit("meter");
+
+    EXPECT_TRUE(u1->equal(u2));
+}
+
+TEST(Equality, unitsEqualAllTestedConditionsVariant2)
+{
+    libcellml::UnitsPtr u1 = libcellml::Units::create("unitsA");
+    libcellml::UnitsPtr u2 = libcellml::Units::create("unitsA");
+
+    libcellml::ImportSourcePtr is = libcellml::ImportSource::create();
+
+    u1->setImportReference("some_location");
+    u2->setImportReference("some_location");
+
+    u1->setSourceUnits(is, "ampere");
+    u2->setSourceUnits(is, "ampere");
+
+    u1->addUnit("second", 1.0, "unit_id");
+    u2->addUnit("second", 1.0, "unit_id");
+
+    EXPECT_TRUE(u1->equal(u2));
+}
+
+TEST(Equality, unitsNotEqualDifferingNumberOfUnits)
+{
+    libcellml::UnitsPtr u1 = libcellml::Units::create("unitsA");
+    libcellml::UnitsPtr u2 = libcellml::Units::create("unitsA");
+
+    libcellml::ImportSourcePtr is = libcellml::ImportSource::create();
+
+    u1->addUnit("second", 1.0, "second_id");
+    u1->addUnit("meter", -1.0, "meter_id");
```

Comment:
these don't exist? Doesn't change the outcome of the test, but it may not be testing what you think it is?

Reply:
For me it is testing what I want it to test.

Reply:
`meter` becomes a user-defined base unit, NOT the built-in `metre` unit.  I think you should change the name to be more obvious it's not the built in unit, if that's what you intended, or correct the spelling so that it is?